tests/test_http_writer.py

```diff
@@ -262,6 +262,18 @@ async def test_write_to_closing_transport(
         await msg.write(b"After closing")
 
 
+async def test_write_to_closed_transport(
+    protocol: Any, transport: Any, loop: Any
+) -> None:
+    msg = http.StreamWriter(protocol, loop)
+
+    await msg.write(b"Before transport close")
+    protocol.transport = None
```

Comment:
The only place I can see where transport is set to `None` is in `write_eof()`. I'm pretty sure this function is not going to be called when the client connection has dropped...

Reply:
Its called here https://github.com/aio-libs/aiohttp/blob/master/aiohttp/base_protocol.py#L64

Reply:
I added a test for this and tweaked it a bit so there is a single source of truth for the transport

Reply:
Right, it's a different object. So, `self._protocol.transport` is set to `None`, but `self._transport` is not.

I'd still like to know the root cause, as to why that gets set to `None` on a disconnection, but the actual transport object still returns `False` from `.is_closing()`. Seems like they should both happen...

Reply:
I expect it has something to do with all the ssl sends being buffered and the plaintext transport tries to send immediately so it knows the connection has dropped.

Reply:
I've tried to track down the issue in the code, but not figured it out.
Seems like _SSLProtocolTransport.is_closing() is the one being used when doing SSL, but it seems to wrap the non-SSL object, so the regular transport would be available under _SSLProtocolTransport._ssl_protocol._transport, and I'm wondering if it should be proxying the call to that transport, which would then given us the same result as non-SSL connections.

Reply:
OK, I don't think we're going to work that out. But, having another look, I think I've found a clean solution.

It seems that `BaseProtocol.connection_lost()` is called reliably and sets an attribute:
https://github.com/aio-libs/aiohttp/blob/master/aiohttp/base_protocol.py#L62

Maybe we should just add a property method to expose that publically, and check for that attribute instead?

Reply:
90f42571b78cea86165adbab392218a7104d0d60